src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs

```diff
@@ -89,115 +89,22 @@ public static int Main()
     }
 
     static public int M(int p)
-    {
+    {|method:{|closure:{
```

Comment:
>method [](start = 6, length = 6)

I wonder if using XML tags wouldn't be a bit easier to read:

```
static public int M(int p)
    <METHOD><CLOSURE>{
        Func<int, int> f1 = delegate(int x)
        <CLOSURE><LAMBDA>{
            int q = 2;
            Func<int, int> f2 = (y) => 
            <LAMBDA>{
                return p + q + x + y;
            }</LAMBDA>;
            return f2(3);
        }</LAMBDA></CLOSURE>;
        return f1(4);
    </CLOSURE></METHOD>}
```

Reply:
The exact syntax for the markers is definitely something to discuss. I used the markup style from the existing IDE tests, but I agree readability suffers (especially as most EnC tests would presumably have at least one marker in the same location as an opening brace) and also it adds unnecessary complexity. This is true of XML too though, as strictly speaking the end tags for the method and for lambdas is unnecessary/indeterminate for these tests.

Reply:
The ending tags are necessary, but only for scoping lambdas - you need to know which closure a lambda belongs to. Alternatively, we can use numbered markers.

Reply:
We considered <| early on, but it's worse when used in a test based of of xml literals (which we have tons of). Note that this {| form has been used a long time (looking predating even our release to open source)

Reply:
EnC tests are not using XML literals, so that's not a problem.